### PR TITLE
Add SlackOptionsResponse model for Dialog Options handling

### PR DESCRIPTION
**Pull request by zmarushchak-hs**
SlackOptionsResponse model is needed to load Options to [Selects](https://api.slack.com/dialogs#select_elements) on [Slack Dialogs](https://api.slack.com/dialogs).

### CODE REVIEW

File: slack-base/src/main/java/com/hubspot/slack/client/models/interaction/BlocksLoadOptionsResponseIF.java [hunk 1]
```diff
@@ -19,6 +20,13 @@
 @HubSpotStyle
 @JsonNaming(SnakeCaseStrategy.class)
 public interface BlocksLoadOptionsResponseIF extends BaseSlackOptionsResponse {
+  String TYPE = "blocks_load_options_response";
+
+  @Derived
+  default String getType() {
+    return TYPE;
+  }
```
Comment (omotnyk) [inline slack-base/src/main/java/com/hubspot/slack/client/models/interaction/BlocksLoadOptionsResponseIF.java:28]: Why do we need this?

File: slack-base/src/main/java/com/hubspot/slack/client/models/interaction/SlackOptionsResponseIF.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+public interface SlackOptionsResponseIF extends BaseSlackOptionsResponse {
+  String TYPE = "slack_options_response";
+
+  // NOTE: slack annoyingly has a different model for dialog options and attachment options
+  // https://api.slack.com/dialogs | https://api.slack.com/docs/message-menus
+
+  @Derived
+  default String getType() {
+    return TYPE;
+  }
```
Comment (omotnyk) [inline slack-base/src/main/java/com/hubspot/slack/client/models/interaction/SlackOptionsResponseIF.java:36]: Why do we need this?

**Review by omotnyk — APPROVED** [review]
Two questions. The rest looks good to me.

File: slack-base/src/main/java/com/hubspot/slack/client/models/interaction/BlocksLoadOptionsResponseIF.java (hunk 1 above)
Comment (zmarushchak-hs) [reply slack-base/src/main/java/com/hubspot/slack/client/models/interaction/BlocksLoadOptionsResponseIF.java:28]: Looks like we didn't. I thought it is used by Jackson to determine type but looks like I was wrong. Removing this.

**Pull request merged**